sky/utils/dag_utils.py

```diff
@@ -94,18 +94,31 @@ def maybe_infer_and_fill_dag_and_task_names(dag: dag_lib.Dag) -> None:
 
 def fill_default_spot_config_in_dag_for_spot_launch(dag: dag_lib.Dag) -> None:
     for task_ in dag.tasks:
-        assert len(task_.resources) == 1, task_
-        resources = list(task_.resources)[0]
-
-        change_default_value: Dict[str, Any] = {}
-        if resources.use_spot_specified and not resources.use_spot:
-            logger.info(
-                'Field `use_spot` is set to false but a managed spot job is '
-                'being launched. Ignoring the field and proceeding to use spot '
-                'instance(s).')
-        change_default_value['use_spot'] = True
-        if resources.spot_recovery is None:
-            change_default_value['spot_recovery'] = spot.SPOT_DEFAULT_STRATEGY
-
-        new_resources = resources.copy(**change_default_value)
-        task_.set_resources({new_resources})
+
+        new_resources_list = []
+        for resources in list(task_.resources):
+            change_default_value: Dict[str, Any] = {}
+            if resources.use_spot_specified and not resources.use_spot:
+                logger.info(
+                    'Field `use_spot` is set to false but a managed spot job is '  # pylint: disable=line-too-long
+                    'being launched. Ignoring the field and proceeding to use spot '  # pylint: disable=line-too-long
+                    'instance(s).')
+            change_default_value['use_spot'] = True
+            if resources.spot_recovery is None:
+                change_default_value[
+                    'spot_recovery'] = spot.SPOT_DEFAULT_STRATEGY
```

Comment:
We probably want to make sure the spot recovery config for all the resources are the same.